<h1 style='color:#0000ec'>2 &nbsp; Modelos de Malthus, Verhulst e Montroll</h1>

<h2 style='color:#0000b7' >&nbsp; 2.1 &nbsp;O modelo de Malthus</h2>

&nbsp; &nbsp; &nbsp; O modelo populacional de Malthus é dado por:
​
$$\frac{dP}{dt}=κP$$
​
&nbsp; &nbsp; onde <b>P</b> é o número de habitantes e <b>κ</b> a constante de proporcionalidade. 

<h3 style='color:#b50000' >&nbsp; &nbsp; 2.1.1 &nbsp;Solução da equação diferencial</h3>

&nbsp; &nbsp; &nbsp; &nbsp; Inicialmente, obtemos a solução da equação diferencial por separação de variáveis:

$$\frac{dP}{dt}=κP \Rightarrow \frac{dP}{κP}=dt $$

&nbsp; &nbsp; &nbsp; &nbsp; Integrando a equação:

$$ \int\frac{dP}{P} = \int κdt \Rightarrow ln|P|+C_{1} = κt + C_{2} \Rightarrow ln|P| = κt + C $$

&nbsp; &nbsp; &nbsp; &nbsp;  Isolando <b>P</b>,temos:

$$ P = e^{κt + C}$$ 

&nbsp; &nbsp; &nbsp; &nbsp;  Tomando $e^{C}$ como a população inicial $P_{0}$ temos a solução analítica do problema de contorno:

$$ P(t) = P_{0}\:e^{κt}$$ 

<h3 style='color:#b50000' >&nbsp; &nbsp; 2.1.2 &nbsp;Solução do problema de contorno</h3>

&nbsp; &nbsp; &nbsp; &nbsp;  Em $t=0$ a população era de 186 488 habitantes, logo:

$$ P(0)=P_{0}e^{κ(0)} \Rightarrow 186 \: 488 = P_{0} $$

&nbsp; &nbsp; &nbsp; &nbsp;  Dessa forma, $P(t)$:

$$ P(t) = 186488\:e^{κt}$$

&nbsp; &nbsp; &nbsp; &nbsp;  Mas, em $t=10$ a população era de 197 228, então:

$$ P(10) = 186488\:e^{(10)κ} \Rightarrow 197228 = 186488\:e^{10κ} \Rightarrow \frac{197228}{186488}=e^{10κ} \Rightarrow $$ <br> 

$$ \Rightarrow ln \left ( \frac{197228}{186488} \right ) = 10κ \Rightarrow κ=\frac{1}{10}ln \left ( \frac{197228}{186488} \right ) \approx 0.0056$$

&nbsp; &nbsp; &nbsp; &nbsp;  Portanto, segundo o modelo de Malthus, a equação que descreve o número de habitantes de Rio Grande - RS sob as condições do <br>  &nbsp; &nbsp; &nbsp; enunciado é dada por:

$$ P(t)=186488e^{0.0056t}$$



<h2 style='color:#0000b7' >&nbsp; 2.2 &nbsp;O modelo de Verhulst</h2>


&nbsp; &nbsp; &nbsp; <b style='color:#b50000'>Referência:</b> https://repositorio.ufsc.br/bitstream/handle/123456789/203170/TCC%20Geraldo%20Campos.pdf?

&nbsp; &nbsp; &nbsp; O modelo populacional de Verhulst é dado por:

$$\frac{dP}{dt}=\left [  κ\left ( 1- \frac{P}{L} \ \right ) \right ]P$$

&nbsp; &nbsp; &nbsp; Onde <b>P</b> é o número de habitantes, <b>L</b> é um limite estipulado em decorrência da competitividade, e, <b>κ</b> a constante de crescimento de uma <br> &nbsp; &nbsp; população em determinada época.

<h3 style='color:#b50000' >&nbsp; &nbsp; 2.1.1 &nbsp;Solução da equação diferencial</h3>

&nbsp; &nbsp; &nbsp; &nbsp;  A solução analítica do modelo de Verhulst é dada por:

$$P(t)=\frac{L P_{0}e^{κt}}{L+P_{0}(e^{κt}-1)}$$

<h3 style='color:#b50000' >&nbsp; &nbsp; 2.1.2 &nbsp;Solução do problema de contorno</h3>

&nbsp; &nbsp; &nbsp; &nbsp;  Podemos escrever o problema enunciado na <b>Seção 1</b> como um problema de contorno:

$$\begin{cases} \frac{dP}{dt}= \left [  κ\left ( 1- \frac{P}{L} \ \right ) \right ]P  \\ P(0)= 186 \: 488 \\ P(10)= 197 \: 228 \end{cases}$$

&nbsp; &nbsp; &nbsp; &nbsp;   Para o modelo de Verhuslt devemos estudar um limite <b>L</b>, isto é, estipular o número de habitantes de Rio Grande após um determinado <br> &nbsp; &nbsp; &nbsp; período. Vamos considerar <b>L = 300 000</b>

 &nbsp; &nbsp; &nbsp; &nbsp;   Em $t=10$ temos $N=197 \: 228$:
 
 $$P(10)=\frac{300 \: 000 * 186 \: 488*e^{κ(10)}}{300 \: 000+186 \: 488(e^{κ(10)}-1)} = 197 \: 228 \Rightarrow κ = ln\left ( \frac{699617023^{1/10}}{598929523^{1/10}} \right ) $$

&nbsp; &nbsp; &nbsp; &nbsp; Calculando κ:

In [1]:
import math
k = math.log(699617023**(1/10)/598929523**(1/10))
print('k =',k)

k = 0.01553891418938226


&nbsp; &nbsp; &nbsp; &nbsp;Logo, a solução geral do problema:<br>


 $$P(t)=\frac{300 \: 000 * 186 \: 488*e^{κ(t)}}{300 \: 000+186 \: 488(e^{κ(t)}-1)}, \: κ = 0.01553891418938226$$

<h3 style='color:#b50000' >&nbsp; &nbsp; 2.1.3 &nbsp;População estimada para 2021</h3>

&nbsp; &nbsp; &nbsp; &nbsp; A função $P(t)$ em $t=21$:

$$P(21)=\frac{300 \: 000 * 186 \: 488*e^{0.01553891418938226(21)}}{300 \: 000+186 \: 488(e^{0.01553891418938226(21)}-1)} = 208447.5573077047$$

&nbsp; &nbsp; &nbsp; &nbsp; Portanto, a simulação para 2021 é de 208 448 habitantes.

<h2 style='color:#0000b7' >&nbsp; 2.3 &nbsp;O modelo de Montroll</h2>

&nbsp; &nbsp; &nbsp; <b style='color:#b50000'>Referência:</b> https://repositorio.ufsc.br/bitstream/handle/123456789/203170/TCC%20Geraldo%20Campos.pdf?

&nbsp; &nbsp; &nbsp; O modelo populacional de Verhulst é dado por:

$$\frac{dP}{dt}=  κP\left [ 1- \left ( \frac{P}{L} \right )^{b}  \ \right ] $$



<b>Simulação para dados do DATASUS 2010-2011-2012</b>

In [29]:
import numpy as np

def Pn(P0,K,r,vt):
    Pt =  (K * P0 * np.exp (r * vt) ) / (K + P0 * (np.exp (r * vt) - 1))
    return Pt

P0 = 197228
K = 218857
r = 0.04285
vt = np.arange(0,3,1)

print('Vetor Tempo: ',vt)

# 2000-2005-2010 dados = np.array([186544, 195392, 197228]) (DataSus)
dados = np.array([197228, 198049, 198842])

Pt = Pn(P0,K,r,vt)

print(Pt)


for i in range(0,Pt.shape[0],1):
    
    erro_perc = ((dados[i] - Pt[i])/dados[i])*100
    print('Dados reais: {} , Dados estimados: {}, Erro percentual : {}%'.format(dados[i],Pt[i],erro_perc))
    
P21 = Pn(P0,K,r,9)
err21 = ((212881 -P21)/212881)*100
print('_________\n Simulação para 2021: ',P21)
print('Erro para 2021: {}%'.format(err21))

Vetor Tempo:  [0 1 2]
[197228.         198048.97112335 198841.94301014]
Dados reais: 197228 , Dados estimados: 197228.0, Erro percentual : 0.0%
Dados reais: 198049 , Dados estimados: 198048.97112335352, Erro percentual : 1.4580556569362828e-05%
Dados reais: 198842 , Dados estimados: 198841.94301014344, Erro percentual : 2.866087474431193e-05%
_________
 Simulação para 2021:  203668.78697678153
Erro para 2021: 4.327400295572866%


<b>Simulação para dados do IBGE 1991-2000-2010</b> (considerando intevalos igualmente espaçados)

In [33]:
import numpy as np

def Pn(P0,K,r,vt):
    Pt =  (K * P0 * np.exp (r * vt) ) / (K + P0 * (np.exp (r * vt) - 1))
    return Pt

P0 = 172422
K = 223012
r = 0.04042
vt = np.array([0,10,20])

print('Vetor Tempo: ',vt)

# 2000-2005-2010 dados = np.array([186544, 195392, 197228]) (DataSus)
#dados = np.array([197228, 198049, 198842])
dados = np.array([172422,186488,197228])

Pt = Pn(P0,K,r,vt)

print(Pt)


for i in range(0,Pt.shape[0],1):
    
    erro_perc = ((dados[i] - Pt[i])/dados[i])*100
    print('Dados reais: {} , Dados estimados: {}, Erro percentual : {}%'.format(dados[i],Pt[i],erro_perc))
    
P21 = Pn(P0,K,r,9)
err21 = ((212881 -P21)/212881)*100
print('_________\n Simulação para 2021: ',P21)
print('Erro para 2021: {}%'.format(err21))

Vetor Tempo:  [ 0 10 20]
[172422.         186487.80809951 197227.65240351]
Dados reais: 172422 , Dados estimados: 172422.0, Erro percentual : 0.0%
Dados reais: 186488 , Dados estimados: 186487.80809950575, Erro percentual : 0.0001029023284321301%
Dados reais: 197228 , Dados estimados: 197227.65240351102, Erro percentual : 0.00017624094397364652%
_________
 Simulação para 2021:  185236.4494615247
Erro para 2021: 12.985917267616792%


<b>Simulação para dados do DATASUS 2000-2005-2010</b>

In [35]:
import numpy as np

def Pn(P0,K,r,vt):
    Pt =  (K * P0 * np.exp (r * vt) ) / (K + P0 * (np.exp (r * vt) - 1))
    return Pt

P0 = 186544
K = 197681
r = 0.004
vt = np.array([0,5,10])

print('Vetor Tempo: ',vt)

# 2000-2005-2010 dados = np.array([186544, 195392, 197228]) (DataSus)
#dados = np.array([197228, 198049, 198842])
#dados = np.array([172422,186488,197228])
dados = np.array([186544, 195392, 197228])

Pt = Pn(P0,K,r,vt)

print(Pt)


for i in range(0,Pt.shape[0],1):
    
    erro_perc = ((dados[i] - Pt[i])/dados[i])*100
    print('Dados reais: {} , Dados estimados: {}, Erro percentual : {}%'.format(dados[i],Pt[i],erro_perc))
    
P21 = Pn(P0,K,r,9)
err21 = ((212881 -P21)/212881)*100
print('_________\n Simulação para 2021: ',P21)
print('Erro para 2021: {}%'.format(err21))

Vetor Tempo:  [ 0  5 10]
[186544.         186752.33566418 186956.99811623]
Dados reais: 186544 , Dados estimados: 186544.0, Erro percentual : 0.0%
Dados reais: 195392 , Dados estimados: 186752.33566417915, Erro percentual : 4.421708327782536%
Dados reais: 197228 , Dados estimados: 186956.99811623126, Erro percentual : 5.2076793780643404%
_________
 Simulação para 2021:  186916.3567658648
Erro para 2021: 12.196787517033087%
